VContainer/Assets/VContainer/Runtime/Annotations/AnnotationUtility.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace VContainer.Unity
+{
+    public static class AnnotationUtility
+    {
+        private static readonly List<Type> _list = new List<Type>
+        {
+#if VCONTAINER_UNITASK_INTEGRATION
+            typeof(IAsyncStartable),
+#endif
+            typeof(IInitializable),
```

Comment:
If we give special treatment to `IInitializable` etc and so on, the behavior when the user uses `IEnumerable<IInitializable>` will be different.

My idea is to have a special type just for this.
For example, how about `ContainerLocalCollection<T>`?

```csharp
sealed class ContainerLocalCollection<T> : IReadOnlyList<T>
{
    public int Count => innerList.Count;
    public T this[int index] => innerList[index];

    public IEnumerator<T> GetEnumerator() => innerList.GetEnumerator();
    IEnumerator IEnumerable.GetEnumerator() => innerList.GetEnumerator();
    
    readonly IReadOnlyList<T> innerList;    
    
    public ContainerLocalCollection(IReadOnlyList<T> innerList)
    {
        this.innerList = innerList;
    }
}
```

Then use it where we want to narrow down the scope (e.g. `EntryPointDispatcher`).

```csharp
var initializables = container.Resolve<ContainerLocalCollection<IInitializable>>();
```

In a few places where we create collections internally, we need to branch if Registration.ImplementationType is ContainerLocalCollection<>. However, I think it will make the intent clearer.